pkg/translator/prometheusremotewrite/metrics_to_prw_v2_test.go

```diff
@@ -0,0 +1,99 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package prometheusremotewrite
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	writev2 "github.com/prometheus/prometheus/prompb/io/prometheus/write/v2"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/pmetric/pmetricotlp"
+)
+
+func TestFromMetricsV2(t *testing.T) {
+	settings := Settings{
+		Namespace:           "",
+		ExternalLabels:      nil,
+		DisableTargetInfo:   false,
+		ExportCreatedMetric: false,
+		AddMetricSuffixes:   false,
+		SendMetadata:        false,
+	}
+
+	ts := uint64(time.Now().UnixNano())
+	payload := createExportRequestWithTimestamp(5, 0, 1, 3, 0, pcommon.Timestamp(ts))
+	want := func() map[string]*writev2.TimeSeries {
+		return map[string]*writev2.TimeSeries{
+			"0": {
+				LabelsRefs: []uint32{1, 2},
+				Samples: []writev2.Sample{
+					{Timestamp: convertTimeStamp(pcommon.Timestamp(ts)), Value: 1.23},
+				},
+			},
+		}
+	}
+	tsMap, symbolsTable, err := FromMetricsV2(payload.Metrics(), settings)
+	wanted := want()
+	require.NoError(t, err)
+	require.NotNil(t, tsMap)
+	require.Equal(t, wanted, tsMap)
+	require.NotNil(t, symbolsTable)
+
+}
+
+func createExportRequestWithTimestamp(resourceAttributeCount int, histogramCount int, nonHistogramCount int, labelsPerMetric int, exemplarsPerSeries int, timestamp pcommon.Timestamp) pmetricotlp.ExportRequest {
+	request := pmetricotlp.NewExportRequest()
```

Comment:
I'm a bit confused with the motivation for this method 🤔 

We're only implementing Gauges, so why do we have something that generates histograms?

Reply:
it's a copy of existing method, I just added support for passing the timestamp, I could potentially just update the existing one.

Reply:
+1 to updating the existing method instead of duplicating it

Reply:
done